### DISCUSSION TITLE
Expected feed generator behavior for the public web view

### DISCUSSION THREAD

**Discussion by emilyliu7321** [bluesky-social/atproto#1933] [Dev Announcements]
**Update as of December 12:** We're going to launch the initial PWI _without_ custom feeds enabled to ensure performance/stability, and also to give devs more time to make tweaks to their feeds! Still excited to make custom feeds public down the line -- will share more info on timelines/dates when we have it 💪 

----

We’re preparing to launch a public web interface soon, and we’re excited for more people to see your custom feeds! Ahead of that, we want to make sure that your feeds will work smoothly to handle both logged-out users and a potential increase in traffic.

**This is the expected behavior from a feed generator:**
- If the feed requires authentication, when it is fetched for a logged-out user, it should return a 401 Unauthorized error.
- Optionally, you can pass a custom message along with the 401 Unauthorized error, which we will show to the user within the Bluesky public web interface in addition to our default copy that explains that they need to be logged in.
- If the feed is receiving too much traffic, regardless of if it is an authed/unauthed request,  it should return a 429 Too Many Requests error. 

Our API client expects `{ error, message }`.

Tentatively, we’re expecting to launch the public web interface on Monday, December 18, in case this date helps you prepare for a potential increase in traffic.  We’ve increased our caching to help mitigate this, and if too much traffic is causing multiple custom feeds to error out, we’ll restrict the public web interface to not show custom feeds.

Feel free to post questions in this thread!

**Comment by emilyliu7321** [comment]
Tagging the creators of some popular feeds here that may be returning 500s (they throw some form of error that our app view doesn’t in-turn know how to handle and turn into a 401) 🙏 

- OnlyPosts, Mutuals (@redsolver)
- Likes, Media (@jcsalterego)
- Like Club, and others (@DavidBuchanan314)
- Mentions (@flicknow)

**Reply by redsolver** [reply]
All the personalized SkyFeed feeds should now return `401 Unauthorized` - just a quick question: If I return a `message` field in the response JSON and that is shown to users, what does the `error` field do?

**Reply by estrattonbailey** [reply]
It's unused by the client for now, but for example in our codebase it loosely corresponds to the name of the error subclass e.g. our `AuthRequiredError` ([source](https://github.com/bluesky-social/atproto/blob/60deea17622f7c574c18432a55ced4e1cdc1b3a1/packages/xrpc-server/src/types.ts#L224), and [mapping](https://github.com/bluesky-social/atproto/blob/main/packages/xrpc/src/types.ts#L48) to `error` values).

**Reply by MarshalX** [reply]
Tagging the creators of some popular feeds examples here
- https://github.com/MarshalX/bluesky-feed-generator (@MarshalX) XDDDD
- https://github.com/mackuba/bluesky-feeds-rb (@mackuba)

**Comment by mackuba** [comment]
Hey! What about an "under maintenance" status, i.e. "I'm migrating my database, everything is under control, the feed will be back shortly, please don't keep pressing Ctrl+R"?

**Reply by devinivy** [reply]
You can serve a 503 http status to indicate that the service is temporarily offline.  The public API will cache errors for a short period of time (on the order of seconds) which should help ease the pain of all the Ctrl+Rs but also ensure when you're back up that folks wont have to wait long to start getting your feed again.

Another option which can be fun is to serve a static response (possibly directly from your load balancer / reverse proxy) that contains a single post explaining to users the status of the feed.

**Comment by emilyliu7321** [comment]
**Update as of December 12:** We're going to launch the initial PWI _without_ custom feeds enabled to ensure performance/stability, and also to give devs more time to make tweaks to their feeds! Still excited to make custom feeds public down the line -- will share more info on timelines/dates when we have it 💪

**Comment by emilyliu7321** [comment]
**Update as of April 12:** Took a while, but feeds are now viewable on the public web interface! This respects the !no-unauthenticated flag, so users with that toggle on will not show up in feeds to logged out users. Let us know if you're experiencing any issues!

**Reply by emilyliu7321** [reply]
Sorry that there really wasn't a heads up here, that's on us.. — tagging some feed creators here to let me know if you come across any issues: @redsolver @rudyfraser @flicknow @jcsalterego
